registry_store.go

```diff
@@ -83,20 +83,14 @@ type RegistryStoreOptions struct {
 
 // RegistryStore is a store that interacts with a remote registry.
 type RegistryStore struct {
-	name             string
 	client           *auth.Client
 	plainHTTP        bool
 	maxBlobBytes     int64
 	maxManifestBytes int64
 }
 
-// NewRegistryStore creates a new [RegistryStore] with the given name and
-// options.
-func NewRegistryStore(name string, opts RegistryStoreOptions) (*RegistryStore, error) {
-	if name == "" {
-		return nil, errStoreNameRequired
-	}
-
+// NewRegistryStore creates a new [RegistryStore] with options.
+func NewRegistryStore(opts RegistryStoreOptions) (*RegistryStore, error) {
```

Comment:
It never returns an error now.

Consider
```suggestion
func NewRegistryStore(opts RegistryStoreOptions) *RegistryStore {
```

Reply:
Updated.